cmd/agent/subcommands/snmp/command.go

```diff
@@ -107,8 +109,43 @@ func Commands(globalParams *command.GlobalParams) []*cobra.Command {
 					port = defaultPort
 				}
 			} else {
+				//If the customer provides only 1 argument : the ip_address
+				//We check the ip address configuration in the agent runtime and we use it for the snmpwalk
 				deviceIP = address
-				port = defaultPort
+				//Allow the possibility to pass the config file as an argument to the command
+				erre := common.SetupConfig(globalParams.ConfFilePath)
+				if erre != nil {
+					fmt.Printf("The config file provided is invalid : %v \n", erre)
+				}
+				snmpConfigList, err := parse.GetConfigCheckSnmp()
```

Comment:
If there is an error, I think we should not continue executing the rest of the code below:

```
instance := parse.GetIPConfig(deviceIP, snmpConfigList)
				if err != nil {
					fmt.Printf("Couldn't parse the SNMP config : %v \n", err)
				}
				if instance.IPAddress != "" {
...
```